src/parse.rs

```diff
@@ -157,7 +159,26 @@ impl<E, T: Iterator<Item=Result<String, E>>> Iterator for Parser<T> {
     }
 }
 
+#[derive(Debug)]
 pub enum Void { }
+
+impl error::Error for Void {
+    fn description(&self) -> &str {
+        "<void>"
+    }
+
+    fn cause(&self) -> Option<&error::Error> {
+        None
+    }
+}
+
+impl fmt::Display for Void {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use std::error::Error;
+        write!(f, "{}", self.description())
+    }
+}
+
```

Comment:
Can we just replace all these function implementations with `unreachable!()`? It's impossible to call any of them.

Reply:
I was actually thinking about using the [void crate](https://crates.io/crates/void). Did you already consider it and decided not to use it for some reason?

Reply:
That would be appropriate! There isn't a good reason for why it wasn't used, mostly I just wasn't sure if it was necessary (you'll notice the `&str` side of things wasn't particularly well fleshed out, as I've only really used/tested it with files/`Read`). Needing impls for `Void` is probably a good sign that we should use the crate instead.